File: src/python/_0019_Remove_Nth_Node_From_End_of_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        first = ListNode(0, head)
        l, r = first, head

        for _ in range(n):
            r = r.next

        while r:
            l, r = l.next, r.next

        l.next = l.next.next

        return first.next

File: src/_0019_Remove_Nth_Node_From_End_of_List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *removeNthFromEnd(struct ListNode *head, int n)
{
    struct ListNode *temp = head;
    int count = 1;
    while (temp->next != NULL)
    {
        temp = temp->next;
        count++;
    }
    if (count == n)
        return head->next;
    temp = head;
    for (int i = 0; i < count - n - 1; i++)
        temp = temp->next;
    temp->next = temp->next->next;
    return head;
}
